docs/env.yml

```diff
@@ -4,15 +4,15 @@ channels:
   - conda-forge
 dependencies:
   - breathe>=4.35
-  - python=3.10
-  - r-base=4.3.2
-  - r-data.table=1.14.10
-  - r-jsonlite=1.8.8
-  - r-knitr=1.45
-  - r-markdown=1.12
-  - r-matrix=1.6_4
-  - r-pkgdown=2.0.7
+  - python=3.12
+  - r-base>=4.3.3
+  - r-data.table=1.16.4
+  - r-jsonlite=1.8.9
+  - r-knitr=1.49
+  - r-markdown=1.13
+  - r-matrix=1.6_5
+  - r-pkgdown=2.1.1
   - r-roxygen2=7.3.2
-  - scikit-learn>=1.4.0
-  - sphinx>=6.0
-  - sphinx_rtd_theme>=2.0
+  - scikit-learn>=1.6.1
+  - sphinx>=8.1.3,<9
+  - sphinx_rtd_theme>=3.0.1,<4
```

Comment:
Why do we need to pin to current major version?

Reply:
I didn't mean to leave this in the diff here, sorry. It shouldn't be part of these changes. I just pushed https://github.com/microsoft/LightGBM/pull/6668/commits/a78ebbddd20caaa6f62caf1d6ac476e9964be130 removing it.